model/codec.go

```diff
@@ -0,0 +1,48 @@
+// Copyright (c) 2021 The Jaeger Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package model
+
+import (
+	proto "github.com/gogo/protobuf/proto"
+	"google.golang.org/grpc/encoding"
+)
+
+func init() {
+	encoding.RegisterCodec(newCodec())
+}
+
+// gogoCodec forces the use of gogo proto marshalling/unmarshalling for
+// proto over grpc
+type gogoCodec struct {
+}
+
+func newCodec() *gogoCodec {
+	return &gogoCodec{}
+}
+
+// Name implements encoding.Codec
+func (c *gogoCodec) Name() string {
+	return "proto"
+}
+
+// Marshal implements encoding.Codec
+func (c *gogoCodec) Marshal(v interface{}) ([]byte, error) {
+	return proto.Marshal(v.(proto.Message))
+}
+
+// Unmarshal implements encoding.Codec
+func (c *gogoCodec) Unmarshal(data []byte, v interface{}) error {
+	return proto.Unmarshal(data, v.(proto.Message))
+}
```

Comment:
I assume this is already explicitly covered by the existing tests for marshalling.

Reply:
Yes, all grpc/proto marshalling/unmarshalling will now go through this file.